Automattic/woocommerce-payments
# Payment Request Button: Guide users to authentication when it is required

Fixes #1502

#### Changes proposed in this Pull Request

- Display the button instead of hiding it when authentication is required.
- Add a dialog which tells users to login before checking out.
- Redirect users back to the checkout flow after authentication.

#### Testing instructions

First, run `npm start` to build the assets from this branch.

1. Go to **WooCommerce > Settings > Accounts & Privacy** and disable the following settings:
  - Allow customers to place orders without an account.
  - Allow customers to create an account during checkout.
2. Make sure you have **Payment request buttons** enabled under **Payments > Settings**.
3. Make sure you're not logged in.
3. Go to a product page in Chrome and click the payment request button.
4. Notice there's a dialog and the "Confirm" button redirects you to my-account.
5. Sign in or create an account.
6. After authentication, you should be redirected back to the same page.

![gif](https://user-images.githubusercontent.com/5509901/122165183-3a8ca980-ce4e-11eb-9d18-b328edc74935.gif)


-------------------
